go/types/value.go

```diff
@@ -8,15 +8,31 @@ import (
 	"github.com/attic-labs/noms/go/hash"
 )
 
-// Value is implemented by every noms value
+// Value is the interface all Noms values implement.
 type Value interface {
+	// Equals determines if two different Noms values represents the same underlying value.
 	Equals(other Value) bool
+
+	// Less determines if this Noms value is less than another Noms value.
+	// When comparing two Noms values and both are comparable and the same type (Bool, Number or
```

Comment:
Should we document the details of the implementation here or maybe just leave that to the documentation of the implementations of Less?